### ISSUE TITLE
Add a rule to keep `aspect-ratio` values consistent

### ISSUE THREAD

**Issue by MorevM** [stylelint-stylistic/stylelint-stylistic#17]
### What is the problem you're trying to solve?

Hello,

I would like to see a new rule to control the appearance of the values of the `aspect-ratio` property.

Without control, the values can be variable, the following entries are all valid:

```css
.block {
  aspect-ratio: 2 / 1;
  aspect-ratio: 2/1;
  aspect-ratio: 2;
  aspect-ratio: 2 / 1 auto;
  aspect-ratio: 2/1          auto;
  aspect-ratio: auto 2 / 1;
}
```

### What solution would you like to see?

I'd like to have control over spaces between key values (around `/`) and over spaces between the parts of the value in a long form (using "auto" and the value both).

It also seems like a good idea to have a setting that disallows shortened entry like `aspect-ratio: .5` because this requires keeping in mind that if there is no value for height, it defaults to `1`. 

It would also be good to make a setting for the position of the keyword "auto" - before the value or after.

I would create this rule.

Now I propose to discuss the name, the settings and its defaults.
I suggest we start with `aspect-ratio-consistent` (I don't like, but can't think of a better one) and settings in form of:
```ts
type Settings = {
  // `1 / 1` vs `1/1`
  spaceWithinRatio: 'always' | 'never',
  
  // `1 / 1 auto` vs `1 / 1        auto`
  singleSpaceBetweenValues: boolean;

  // `auto 1 / 1` vs `1 / 1 auto`
  autoKeywordPosition: 'before' | 'after'
}
```

Maybe the option `singleSpaceBetweenValues` can be enabled by the primary option 🤔
It's hard for me to imagine a scenario where someone would want to turn it off..

**Comment by silverwind** [comment]
A rule for spaces around slashes might be nice. Also applies to this newish color syntax:

```css
:root {
  --color: rgb(255 0 0 / 80%);
}

**Comment by firefoxic** [comment]
@MorevM Hi!
I think we need three rules:

1. Controlling the spaces around `/` in all values. This is not just for `aspect-ratio` or modern color function notation. Here are some examples where you'd also like to put whitespace around `/`:

	```css
	.selector {
		grid:
			" a   b   c  "
			" d   e   f  "
			/ 1fr 2fr 3fr;
		grid-area: 1 / 2 / 3 / 4;
		background: url("icon.svg") no-repeat center / contain;
		font: 20px / 1.5 system-ui, sans-serif;
	}
	```

	And even a possible future [delimiter for logical values](https://github.com/w3c/csswg-drafts/issues/1282#issuecomment-1250032190):

	```css
	.selector {
		margin: 1em 2em / 3em 4em;
		padding: 1em 2em / 3em 4em;
	}
	```

3. Disallowing a shortened entry for `aspect-ratio`. Although technically it probably works as described in the documentation, i.e. the missing value for height is taken as `1`. I could be wrong, but I think that in practice few people think this way. Rather, most people think of the singular number simply as a finite value for the `width`/`height` ratio. But a setup like this would definitely be useful to someone (at least beginners and students).

4. Controlling the order of values. Maybe this could be extended beyond aspect ratio to properties like background, mask, transition, animation, and others. I am really not sure if this would be easy to do. Also, for some abbreviations, such as `font`, the specification requires a rather strict order.

**Comment by silverwind** [comment]
> Disallowing a shortened entry for aspect-ratio

Also make that one support the inverse, e.h. prefer `2` over `2 / 1` when second value.

**Comment by MorevM** [comment]
Good catches. Well, let's start with (2) point as it is obviously the simplest one.

I'll create a separate issue for each new rule discussion. 
Let's leave this one open for a while for new opinions on the whole scenario.

@silverwind @firefoxic [Here](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/18) is the one for discussing about `aspect-ratio` shortened value.

**Comment by firefoxic** [comment]
> it is obviously the _simplest_ one.

I wouldn't rush to that assessment 😅
I'll come back with my thoughts soon. It doesn't seem to be that simple there.
